crates/solvers/src/domain/solver/legacy.rs

```diff
@@ -27,10 +27,13 @@ impl Legacy {
     }
 
     pub async fn solve(&self, auction: auction::Auction) -> Vec<solution::Solution> {
-        match self.0.solve(auction).await {
+        match self.0.solve(&auction).await {
             Ok(solution) => vec![solution.with_id(solution::Id(0))],
             Err(err) => {
                 tracing::warn!(?err, "failed to solve auction");
+                if err.is_timeout() {
```

Comment:
nit: If `.is_timout()` is only used in one place it might be nicer to not have the specialized function at all.
Feel free to ignore, though.
```suggestion
                if matches!(err, Error::Timeout) {
```

Reply:
Yes, I thought one would need less cognitive effort to read `is_timeout` then the matches! version 😄 . But don't feel strongly about it.